patches/codehook.slpatch

```diff
@@ -1,8 +1,3 @@
 [version=310, target=main]
-gsys::SystemTask::invokeDrawTV_+284 NOP // enable display debug stuff (which is used for hook)
-gsys::SystemTask::invokeDrawTV_+390:
-	MOV X1, X0
-	MOV X0, X25
-	BL render
-	B #0x2B8
-	
\ No newline at end of file
+StageScene::control+18:
```

Comment:
This line means that our instruction replacements will start `0x18` bytes into the `StageScene::control` function.

Starlight will use the `main.map` file (also updated in this PR) to look up the starting address of this function (`0x71004CC348`), add `0x18` to it (`0x71004CC360`), and start the instruction replacements there.

Since we're only replacing a single instruction, this is equivalent to a .pchtxt setup of

```
004CC360 XXXXXXXX
```

(where `XXXXXXXX` is whatever the hex representation of our instruction ends up being)

Reply:
To be even more specific, this is replacing the first instruction after the function's prologue: `MOV X19, X0`.

While not true of every replacement, in our case we'll need to keep this original instruction in mind so we can re-implement it elsewhere — instructions later in the function rely on `X19` having the correct value, and overwriting this `MOV` instruction means that this won't be the case at present.